public/app/plugins/datasource/elasticsearch/language_provider.ts

```diff
@@ -0,0 +1,110 @@
+  const openBracketIndex = promQuery.indexOf('{');
+  const lastOpenParenthesis = promQuery.lastIndexOf('(');
+  var startNameLabelIndex = 0;
+  if (lastOpenParenthesis !== -1) {
+    startNameLabelIndex = lastOpenParenthesis + 1;
+  }
+  let lastNameLabelIndex = openBracketIndex;
+  if (openBracketIndex === -1) {
+    const openBraceIndex = promQuery.indexOf('[');
+    if (openBraceIndex === -1) {
+      lastNameLabelIndex = promQuery.length;
+    } else {
+      lastNameLabelIndex = openBraceIndex;
+    }
+  }
+  let nameLabelValue = promQuery.substring(startNameLabelIndex, lastNameLabelIndex);
+  return nameLabelValue;
+}
+async function extractPrometheusLabels(promQuery: string): Promise<string[][]> {
+  var labels: string[][] = [];
+  if (!promQuery || promQuery.length === 0) {
+    return labels;
+  }
+  const nameLabelValue = await getNameLabelValue(promQuery);
+  if (nameLabelValue && nameLabelValue.length > 0) {
+    labels.push(['__name__', '=', '"' + nameLabelValue + '"']);
+  }
+
+  let m;
+  while ((m = labelRegexp.exec(promQuery)) != null) {
+    labels.push([m[1], m[2], m[3]]);
+  }
+  return labels;
+}
+async function getElasticsearchQuery(prometheusLabels: string[][]): Promise<string> {
+  var elasticsearchLuceneLabels = [];
+  for (let keyOperatorValue of prometheusLabels) {
+    switch (keyOperatorValue[1]) {
+      case '=': {
+        elasticsearchLuceneLabels.push(keyOperatorValue[0] + ':' + keyOperatorValue[2]);
+        break;
+      }
+      case '!=': {
+        elasticsearchLuceneLabels.push('NOT ' + keyOperatorValue[0] + ':' + keyOperatorValue[2]);
+        break;
+      }
+      case '=~': {
+        elasticsearchLuceneLabels.push(
+          keyOperatorValue[0] + ':/' + keyOperatorValue[2].substring(1, keyOperatorValue[2].length - 1) + '/'
+        );
+        break;
+      }
+      case '!~': {
+        elasticsearchLuceneLabels.push(
+          'NOT ' + keyOperatorValue[0] + ':/' + keyOperatorValue[2].substring(1, keyOperatorValue[2].length - 1) + '/'
+        );
+        break;
+      }
+    }
+  }
+  return elasticsearchLuceneLabels.join(' AND ');
+}
+
+export default class ElasticsearchLanguageProvider extends LanguageProvider {
+  request: (url: string, params?: any) => Promise<any>;
+  start: () => Promise<any[]>;
+  datasource: ElasticDatasource;
+
+  constructor(datasource: ElasticDatasource, initialValues?: any) {
+    super();
+    this.datasource = datasource;
+
+    Object.assign(this, initialValues);
+  }
+
+  async importQueries(queries: DataQuery[], datasourceType: string): Promise<ElasticsearchQuery[]> {
+    if (datasourceType === 'prometheus' || datasourceType === 'loki') {
+      return Promise.all(
+        queries.map(async query => {
+          let prometheusQuery: PromQuery = query as PromQuery;
+          const expr = await getElasticsearchQuery(await extractPrometheusLabels(prometheusQuery.expr));
+          return {
+            isLogsQuery: true,
```

Comment:
This could be a bit problematic because we do not actually change the mode in Explore based on this. So if you go from Prometheus to Elastic it will stay in Metrics mode but the queries will be done as if in Logs mode which may result in errors, though we have this issue without this PR. @kaydelaney does it make sense to do something about it or will this be moot anyway in 7.1 when we remove the mode switcher?

Reply:
I don't get what do you mean, you can see the logs when you switch from prometheus to elasticsearch.

Reply:
Hey @aocenas 
Is it a blocker to not merge the PR please ?